files/Pvsamv1Tools.py

```diff
@@ -0,0 +1,376 @@
+    :param n_inputs_combiner: number of DC inputs per combiner box
+
+    :returns: number of strings, number of combiner boxes, number of inverters, calculated system capacity, kW
+    """
+    
+
+    module_model = model.Module.module_model
+    module_power = 0
+    module_vmp = 0
+    module_voc = 0
+    match module_model:
+        case 0:
+            ref = int(model.SimpleEfficiencyModuleModel.spe_reference)
+            eff_list = [model.SimpleEfficiencyModuleModel.spe_eff0, model.SimpleEfficiencyModuleModel.spe_eff1, model.SimpleEfficiencyModuleModel.spe_eff2, model.SimpleEfficiencyModuleModel.spe_eff3, model.SimpleEfficiencyModuleModel.spe_eff4]
+            rad_list = [model.SimpleEfficiencyModuleModel.spe_rad0, model.SimpleEfficiencyModuleModel.spe_rad1, model.SimpleEfficiencyModuleModel.spe_rad2, model.SimpleEfficiencyModuleModel.spe_rad3, model.SimpleEfficiencyModuleModel.spe_rad4]
+            eff = eff_list[ref]
+            rad = rad_list[ref]
+            area = model.SimpleEfficiencyModuleModel.spe_area
+            module_power = eff / 100.0 * rad * area #Wdc
+            module_vmp = model.SimpleEfficiencyModuleModel.spe_vmp
+            module_voc = model.SimpleEfficiencyModuleModel.spe_voc
+        case 1:
+            module_power = model.CECPerformanceModelWithModuleDatabase.cec_v_mp_ref * model.CECPerformanceModelWithModuleDatabase.cec_i_mp_ref
+            module_vmp = model.CECPerformanceModelWithModuleDatabase.cec_v_mp_ref
+            module_voc = model.CECPerformanceModelWithModuleDatabase.cec_v_oc_ref
+        case 2:
+            module_vmp = model.CECPerformanceModelWithUserEnteredSpecifications.sixpar_vmp
+            module_voc = model.CECPerformanceModelWithUserEnteredSpecifications.sixpar_voc
+            module_power = module_vmp * model.CECPerformanceModelWithUserEnteredSpecifications.sixpar_imp
+        case 3:
+            print('This function does not currently work for the Sandia Array Performance Model')
+            '''
+            module_power = model.SandiaPVArrayPerformanceModelWithModuleDatabase.snl_ref_pmp
+            module_vmp = model.SandiaPVArrayPerformanceModelWithModuleDatabase.snl_ref_vmp
+            module_voc = model.SandiaPVArrayPerformanceModelWithModuleDatabase.snl_ref_voc
+            '''
+        case 4:
+            module_vmp = model.IEC61853SingleDiodeModel.sd11par_Vmp0
+            module_voc = model.IEC61853SingleDiodeModel.sd11par_Voc0
+            module_power = module_vmp * model.IEC61853SingleDiodeModel .sd11par_Imp0
+        case 5:
+            module_vmp = model.MermoudLejeuneSingleDiodeModel.mlm_V_mp_ref
+            module_voc = model.MermoudLejeuneSingleDiodeModel.mlm_V_oc_ref
+            module_power = module_vmp * model.MermoudLejeuneSingleDiodeModel.mlm_I_mp_ref
+
+    inverter_model = model.Inverter.inverter_model
+    inv_power = 0
+    vdcmax_inv = 0
+    v_mppt_max = 0
+    v_mppt_min = 0
+    match inverter_model:
```

Comment:
I didn't realize match (switch) case was only added to Python as of version 3.10, so I will have to update this to an if loop as we still support Python 3.8 and up in Pypi.

Reply:
We're going to support Python 3.9 until its end-of-life in October 2025